index.js

```diff
@@ -94,6 +94,24 @@ function TChannel(options) {
 }
 require('util').inherits(TChannel, require('events').EventEmitter);
 
+// Decoulping config and creation from the constructor.
+// This also allows us to better unit test the code as the test process
+// is not blocked by the listening connections
+TChannel.prototype.createTchannel = function () {
```

Comment:
Let's call this `.createServer()` or something to imply that it is creating a TCP server.